homeassistant/components/recswitch.py

```diff
@@ -0,0 +1,39 @@
+"""
+Support for Ankuoo RecSwitch MS6126 devices.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/switch.recswitch/
+"""
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_PORT
+from homeassistant.helpers import config_validation as cv
+
+REQUIREMENTS = ['pyrecswitch==1.0.1']
+
+DOMAIN = 'recswitch'
+DATA_RSH = 'RSH'
+
+_LOGGER = logging.getLogger(__name__)
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Optional(CONF_PORT): cv.positive_int,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+async def async_setup(hass, base_config):
+    """Setup recswitch network and start datagram endpoint."""
+    config = base_config.get(DOMAIN, dict())
```

Comment:
```py
config = base_config[DOMAIN]
```

Reply:
ok, but this change willl make the component config section required. I mean....

recswitch:

switch01:
  - platform: recswitch
    name: Garden Lights
    host: 192.168.X.X
    mac: 'F0:FE:6B:XX:XX:XX'

It should be right? or I miss something?
actually, I read more more than one time the developer doc but I'm still unsure about if I really need to create a component and a platform for my devices..

I found this way to share the udp socket server across more platform instances. This is required because all this devices send udp message to the same port to the host.

Suggestions are welcome!   
Thanks

Reply:
If we want to share some device connection between platform types a component is required. We are also moving to use config entries instead as an alternative to the config sections in yaml, and then a component is required.

If you're planning to add more platform types, keep the component. But if there will only be the switch platform it would also work to save the server instance in `hass.data` and reuse it for many switch platform instances.

Reply:
Ok, reusing the server instance for more switch is what I thought as alternative.

I don't known if exists other kind of devices based on this protocol (such as lights or sensors...) I've no connection with the manufactur I've just bought an ankuoo device thinking it was already supported (I misunderstood a thread on the forum ..) so I spent 'some' hours  in reverse engineering.. 

May be sharing the server instance across switch devices is the easiest way.. I'll code in this way.